sql/src/main/java/io/crate/analyze/expressions/ExpressionAnalyzer.java

```diff
@@ -864,9 +849,15 @@ protected Symbol visitSubqueryExpression(SubqueryExpression node, ExpressionAnal
              *
              * However, we support a single column RowType through the SingleColumnTableType.
              */
-            Field field = fields.get(0);
-            SingleColumnTableType singleColumnTableType = new SingleColumnTableType(field.valueType());
-            return new SelectSymbol(relation, singleColumnTableType);
+            DataType innerType = fields.get(0).valueType();
+            SingleColumnTableType dataType = new SingleColumnTableType(innerType);
+            final SelectSymbol.ResultType resultType;
+            if (context.isArrayExpression(node)) {
```

Comment:
If it's somehow possible to create a nesting like the following the logic here is broken as it wouldn't detect the array expression parent correctly. 

```
ArrayComparisonExpression(IntermediateDummyExpression(SubqueryExpression))
```



In other places we usually just use a bool or some kind of stack variable on the context.

```
// in visitArrayComparisonExpression

context.insideArrayComparison = true
process(...)
context.insideArrayComparison = false
```

But I think it's not possible to create any query where there is an
intermediate expression between the array comparsion and the subquery. So I
guess this is okay.